### DISCUSSION TITLE
Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported

### DISCUSSION THREAD

**Discussion by jimsakiko** [codemagic-ci-cd/codemagic-docs#2715] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Using awesome_notifications Flutter package. My Android app built locally works properly, with local notifications as expected.
I built the awesome_notifications_sample app which has reduced the issue to a documented fix if using a Xcode build on a physical machine (which I don't have). According to the a_n FAQ at https://pub.dev/packages/awesome_notifications#common-known-issues , a couple of config adjustments are required, viz:

------------------------------
Issue: Using bridging headers with module interfaces is unsupported

Fix: You need to set build settings options below in your Runner target:

    Build libraries for distribution => NO
    Only safe API extensions => NO

.. and in your Notification Extension target:

    Build libraries for distribution => NO
    Only safe API extensions => YES

------------------------------

I have seen various screenshots where (at least) the "Build Settings" can be tweaked if using Xcode locally, but cannot see how to effect these config selections with the standard CodeMagic workflow.

### Expected results

Clean compile and iOS build. 

I have already deployed a different app through CM all the way through to Test Flight.

### Actual results

Automatically signing iOS for device deployment using specified development team in Xcode project: 642QLF9TS6
Running pod install...                                           2,334ms
Running Xcode build...                                          
Xcode archive done.                                         35.9s
Failed to build iOS app
**Swift Compiler Error (Xcode): Using bridging headers with module interfaces is unsupported**

Encountered error while archiving for device.

### Build id (optional)

66d868adda41cf3702b2768c

**Comment by himesh-cm** [comment]
Hey @jimsakiko,
This seems to be a project-specific issue. Could you please check and update your Flutter packages to the latest versions, then try again?

**Reply by jimsakiko** [reply]
I have now upgraded the Flutter project dependencies as high as possible while still maintaining resolvability. Below is from my local build:

Package Name              Current  Upgradable  Resolvable  Latest

direct dependencies:
collection                *1.18.0  *1.18.0     *1.18.0     1.19.0
http_parser               *4.0.2   *4.0.2      *4.0.2      4.1.0
material_color_utilities  *0.11.1  *0.11.1     *0.11.1     0.12.0
You are already using the newest resolvable versions listed in the 'Resolvable' column.
Newer versions, listed in 'Latest', may not be mutually compatible.

Their minimum sample app continues to work as expected on Android.

However, I get the same compiler error on the Codemagic build ("Using bridging headers with module interfaces is unsupported").

I agree that this is a project-specific issue, but have you looked at the solution indicated by the package publisher that I cited above? Is this not possible on Codemagic?

**Reply by himesh-cm** [reply]
Hey @jimsakiko , Please check if [this](https://stackoverflow.com/questions/74626966/flutter-swift-compiler-error-xcode-using-bridging-headers-with-module-inter) SO thread is helpful.

**Reply by jimsakiko** [reply]
Well, it would if CM was XCode. The last post in that SO thread you linked is _exactly_ what I want to do... as indicated in my just-previous reply.

**Reply by himesh-cm** [reply]
Please share the versions of Flutter and Xcode you use locally. Also, confirm if you are able to build the Flutter iOS app after performing a clean build, i.e., deleting the pub cache and pod cache and rebuilding the app.

**Reply by jimsakiko** [reply]
I'm in a Windows 11 environment using VSCode.

Flutter version 3.24.4-0.0.pre.1 on channel stable at C:\Users\jim\flutter
Dart version 3.5.3
DevTools version 2.37.3

I don't have any kind of Apple -- the reason I am trying to build iOS on Codemagic. Seemed like a viable solution a couple of months ago!

**Comment by icarusdust** [comment]
> I have seen various screenshots where (at least) the "Build Settings" can be tweaked if using Xcode locally, but cannot see how to effect these config selections with the standard CodeMagic workflow.

You can remote access builder machines through SSH/VNC and after making the configuration changes, you can commit the changes using git and push them to your account: https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/
